### DISCUSSION TITLE
Make it easier to start using Uno

### DISCUSSION THREAD

**Discussion by weitzhandler** [unoplatform/uno#10239] [Ideas]
# Feature request

## Title
Please make it more accessible for new users who who want to try out Uno to start using it, by addressing some of the following features:

- [ ] Make sure the generated template project (in vsix) works without warnings or errors out the box, on all platforms
- [ ] More or better project templates, such that cover MVVM using a known library for example
- [ ] Integration in [WindowsTemplateStudio](https://github.com/Microsoft/WindowsTemplateStudio) (See [#2658](https://github.com/Microsoft/WindowsTemplateStudio/issues/2658))
- [ ] Make it easier to port existing UWP projects to Uno, maybe a manual or even a tool
- [x] Official support from a recognized MVVM framework that provides (Completed - [RxUI](https://github.com/reactiveui/ReactiveUI) has now official support for Uno):
  - [x] MVVM support
  - [x] Navigation/routing/modal dialogs
  - [x] DI

## Description
I find the preset Uno templates in Visual Studio to be limited. Please port some more template to Uno, such that support more MVVM platforms (like Prism), and especially such that come with a basic navigation/layout setup.  
The ideal solution would be having Uno projects created with the [WindowsTemplateStudio](https://github.com/Microsoft/WindowsTemplateStudio) repo, but for a start, a basic generated responsive navigation setup would be very nice.  
Would also be worth looking at the [Windows 10](https://github.com/Windows-XAML/Template10) repo.

Additionally, it should be easier to port existing UWP projects to Uno. One way to achieve this can be extending the UWP project-item context-menu so that it offers another option that generates the Android/iOS/WASM projects into the same solution/folder.

**Comment by jeromelaban** [comment]
For now we have not progressed on that front. The Template10 integration in Prism may help having a more complete sample, and WTS is a bit too complex to handle for now (there are too many dependencies).

It will probably be easier for now to have the uno vsix include basic templates for some known frameworks, pre configured.

**Comment by weitzhandler** [comment]
Looks like Prism dropped UWP support so I think it would be better to stick to Rx.
Anyway, even not with WTS, would be nice to integrate in the Uno vsix even a single template that makes use of the following, that will help users get start with Uno easily and encourage devs to start experimenting and making projects with Uno.

The template should consist of:

- MVVM and RxUI based
- Login page at the beginning of app
- Menu + master-detail page
- A simple Todo data entity
- Fluent design
- Minimal responsiveness between mobile and desktop

I'm willing to participate.

**Comment by jeromelaban** [comment]
Would definitely be an interesting addition to the base templates for the vsix templates! If you have a base example that you use for your testing with the ReactiveUI.Uno package, we can start from there.

**Comment by weitzhandler** [comment]
So far, I've been using Prism in all my projects. I'm pretty new to RxUI. I love the way it works, but I find it lacks functionality when it comes to DI, navigation and routing.

So I'm not so sure about how to design the 'Shell' and the overall bootstrapping of the project. TBH, even they recommend creating all constructors as optional with `service = Locator.Current.GetService<IService>()`, personally I don't like this. In prism, once you have DI in control you just worry about the current class and its dependencies, the VM has no knowledge of the container.

If I'm to create that project, I'll do it using `Splat.DryIoc`, since Unity isn't present.

**Comment by weitzhandler** [comment]
And now that integrated RxUI support is almost there it's gonna be easier to update and use other dependent libraries.
I'll just wait until RxUI.Uno is on NuGet. Hope to find time and make a commit within a week.

**Comment by ghuntley** [comment]
ICYMI - The workshop from UnoConf has been open-sourced and contains all of my notes from onboarding/joining the team. It's a fantastic resource for learning how to build applications with Uno and also how to work with the source-code under the hood.

https://github.com/unoplatform/workshops/tree/master/uno-bootcamp

<blockquote><img src="https://repository-images.githubusercontent.com/209543743/63381180-dab7-11e9-8253-6729e5722463" width="48" align="right"><div><img src="https://github.githubassets.com/favicon.ico" height="14"> GitHub</div><div><strong><a href="https://github.com/unoplatform/workshops">unoplatform/workshops</a></strong></div><div>workshops, study guides and learning materials for the Uno Platform  - unoplatform/workshops</div></blockquote>

**Comment by weitzhandler** [comment]
@ghuntley thank you!
